dnf/repo.py

```diff
@@ -554,7 +556,14 @@ def load(self):
         try:
             ret = self._repo.load()
         except RuntimeError as e:
+            if self._md_pload.mirror_failures:
+                msg = "Errors during downloading metadata for repository '%s':" % self.id
+                for failure in self._md_pload.mirror_failures:
+                    msg += "\n  - %s" % failure
+                logger.warning(msg)
             raise dnf.exceptions.RepoError(str(e))
+        finally:
+            self._md_pload.mirror_failures = set()
```

Comment:
Would it be possible to reset the `mirror_failures` in the `MDPayload` class, so that it's self-contained and not dependent on resetting from outside? I'm thinking of putting it into `MDPayload.start()`.

Reply:
Good point and the ``MDPayload.start()`` looks like the right place to put it into. However, it's called in two places in libdnf and can occur twice per one ``load()`` call, with the second one being when ``importRepoKeys()`` is called. So if the ``mirror_failures`` are resetted in ``MDPayload.start()``, a GPG error may hide some other errors.

This happens in ci-dnf-stack repo-sync.feature scenario "Test repo_gpgcheck=1 error if repomd.xml.asc is not present". I haven't found any other reproducer, but even so, I am not sure if it's worth it. What do you think?

Reply:
Huh. We certainly don't want errors to get lost, but the problem is start() being called twice :slightly_frowning_face: it shouldn't be used like that, but I'd guess we don't wanna go fixing it :slightly_frowning_face: 

You could maybe put it in `MDPayload.end()`, if it happens to be called in the right place... Which probably isn't the case. So I suppose keep it this way...

Reply:
Well, one is for the repository metadata download and the other is for getting the GPG key, so maybe it's intended to be used like that. Not sure. :/

``MDPayload.end()`` is always called right after the download, before the exception is thrown, so that doesn't help.